doc/design.md

```diff
@@ -51,10 +51,58 @@ backend --> customerCluster: manages
 @enduml
 ```
 
+#### EM Provides a REST Interface
+`dsn~rest-interface~1`
+
+EM provides a REST interface to clients.
+
+Covers:
+* [`req~rest-interface~1`](system_requirements.md#em-provides-a-rest-interface)
+
+Needs: impl, itest
+
+#### EM Implemented as a Go Library
+`dsn~go-library~1`
+
+EM is implemented as a Go library instead of an executable Go program.
+
+Rationale:
+This allows embedding EM in other applications like SaaS, so it can be reused in different contexts. Nevertheless EM additionally contains an executable Go program which is useful for testing EM and extensions.
```

Comment:
```suggestion
Rationale:

This allows embedding EM in other applications like SaaS, so it can be reused in different contexts

Comment:
 
Nevertheless EM additionally contains an executable Go program which is useful for testing EM and extensions.
```